nominal/__init__.py

```diff
@@ -1,13 +1,48 @@
-import os
+from .nominal import (
+    set_default_connection,
+    get_default_connection,
+    upload_dataset_from_pandas,
+    upload_dataset_from_polars,
+    upload_dataset,
+    get_dataset_by_rid,
+    update_dataset,
+    create_run,
+    get_run_by_rid,
+    search_runs,
+    update_run,
+    add_dataset_to_run,
+    list_datasets_for_run,
+    add_attachment_to_run,
+    list_attachments_for_run,
+    upload_attachment,
+    get_attachment_by_rid,
+    update_attachment,
+    save_attachment,
+)
+from .sdk import Dataset, Run, Attachment, NominalClient
 
-# Allows:
-# import nominal as nm
-# nm.cloud ...
-# nm.data ...
-from . import cloud as cloud
-from . import data as data
-from .nominal import Dataset as Dataset
-from .nominal import Ingest as Ingest
-from .nominal import Run as Run
-
-os.environ["NOMINAL_BASE_URL"] = "https://api-staging.gov.nominal.io/api"
+__all__ = [
+    "set_default_connection",
+    "get_default_connection",
+    "upload_dataset_from_pandas",
+    "upload_dataset_from_polars",
+    "upload_dataset",
+    "get_dataset_by_rid",
+    "update_dataset",
```

Comment:
thoughts on making the function-based version of this more targeted? e.g `update_dataset_name` / etc for the fields we want people to think of as mutable

(also worth noting that there are a bunch of catalog endpoints that ideally wouldn't be public at all - such as updating bounds, ingest status, etc, to lets make sure they're excluded from both python apis)

Reply:
oh interesting, didn't think about that. we could do that, but might be kinda polluting the namespace? I'm not sure. users can always do each one individually:

```py
update_dataset(dataset, name="new name")
update_dataset(dataset, description="new description")
```

Reply:
summary from async chat: 
- maybe not necessary to have an update flow as part of the function-oriented api
- so long as fields like labels/properties can be set on creation, updates can be part of the object-oriented version (I.e `dataset.update(description="new description")` would be the way to do an update)